Sources/SmithyIdentity/AWSCredentialIdentity.swift

```diff
@@ -14,6 +14,7 @@ import struct Foundation.Date
 public struct AWSCredentialIdentity: Identity {
     public let accessKey: String
     public let secret: String
+    public let accountID: String?
```

Comment:
`accountID` is added to AWS credentials (per the account ID-based endpoints SEP).  AWS credential identity resolvers will fill `accountID` when it is available.